ajax/savedsearch.php

```diff
@@ -92,8 +93,23 @@
       $_GET['type']  =(int)$_GET['type'];
    }
 
+   $id = 0;
+   $saved_search = new SavedSearch();
+
+   // If an id was supplied in the query and that the matching saved searche
+   // is private OR the current user is allowed to edit public searches, then
+   // pass the id to showForm
+   if ($saved_search->getFromDB($_GET['id'] ?? 0)) {
```

Comment:
You could prevent a useless query.
```suggestion
   if (($requested_id = $_GET['id'] ?? 0) > 0 && $saved_search->getFromDB($requested_id)) {
```